renku/core/management/migrations/m_0009__new_metadata_storage.py

```diff
@@ -372,6 +373,24 @@ def _process_workflows(client: LocalClient, activity_gateway: IActivityGateway,
 
 def _process_run_to_new_activity(process_run: old_schema.ProcessRun, client: LocalClient) -> List[Activity]:
     """Convert a ProcessRun to a new Activity."""
+
+    def generate_activity_id(process_run_id: str, suffix: str = None) -> str:
+        # https://localhost/activities/commit/b092dca8866c940dd4ff119b7d76d163a1be6a65
+        # https://localhost/activities/commit/b092dca8866c940dd4ff119b7d76d163a1be6a65/steps/step_1
+        parts = process_run_id.rstrip("/").rsplit("/", maxsplit=3)
+
+        if parts[-2] == "steps":
+            parts = parts[-3:-1]
```

Comment:
Is there a reason not to use `parts[-3:]`? It already has the step number in it and there is no need for the suffix.